File: simple/go/part4/internal/interpreteter/token.go
package interpreteter

const (
	INTEGER = "0-9"
	MUL     = "*"
	DIV     = "/"
	NEWLINE = "\\n"
)

type Token interface {
	Type() string
	Value() int
}

type TokenFactory interface {
	NewToken(value int, kind string) Token
}

File: simple/rust/part09/src/token.rs
#[derive(Debug, Clone, PartialEq)]
pub enum Kind {
    INTEGER(i32),
    PLUS,
    MINUS,
    MUL,
    DIV,
    LPAREN,
    RPAREN,
    ID(String),
    ASSIGN,
    BEGIN,
    END,
    SEMI,
    DOT,
    EOF,
}
